pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/bloomfilter/BloomFilterHandler.java

```diff
@@ -136,17 +138,132 @@ private void createBloomFilterForColumn(SegmentDirectory.Writer segmentWriter, C
     BloomFilterConfig bloomFilterConfig = _bloomFilterConfigs.get(columnName);
     LOGGER.info("Creating new bloom filter for segment: {}, column: {} with config: {}", segmentName, columnName,
         bloomFilterConfig);
-    try (BloomFilterCreator bloomFilterCreator = indexCreatorProvider.newBloomFilterCreator(
-        IndexCreationContext.builder().withIndexDir(indexDir).withColumnMetadata(columnMetadata)
-            .build().forBloomFilter(bloomFilterConfig));
-        Dictionary dictionary = getDictionaryReader(columnMetadata, segmentWriter)) {
-      int length = dictionary.length();
-      for (int i = 0; i < length; i++) {
-        bloomFilterCreator.add(dictionary.getStringValue(i));
+
+    if (columnMetadata.hasDictionary()) {
```

Comment:
(nit) suggest separating this into different methods for readability. Something like ....

```
createBloomFilterOnDictColumn
createBloomFilterOnNoDictSVColumn
createBloomFilterOnNoDictMVColumn
```